core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -66,8 +66,13 @@ class TopicConfigHandler(private val logManager: LogManager, kafkaConfig: KafkaC
       topicConfig.asScala.forKeyValue { (key, value) =>
         if (!configNamesToExclude.contains(key)) props.put(key, value)
       }
-      val logConfig = LogConfig.fromProps(logManager.currentDefaultConfig.originals, props)
-      logs.foreach(_.updateConfig(logConfig))
+      val newLogConfig = LogConfig.fromProps(logManager.currentDefaultConfig.originals, props)
```

Comment:
I assume we're doing this here because Log doesn't have access to the log manager? Would it be better to have the comparison logic in the LogManager where it'll be more visible, e.g. maybeAbortCleaning(oldConfig, newConfig)?

Reply:
That is correct. Yeah, that's a good suggestion. I will do this.

Reply:
Another way would be to move `updateLogConfig` in the LogManager. It would make sense to delegate the whole logic to it. What do you think?

Reply:
That does seem better.